### ISSUE TITLE
Quiz Question preview from question bank: Error writing to database: ERROR: value too long for type character varying(32)

### ISSUE THREAD

**Issue by jgramp** [andrewhancox/moodle-filter_translations#79]
Quiz Question preview from question bank: Error writing to database: ERROR: value too long for type character varying(32)

The span tag is making the choices for 'select missing words' questions too long.

![image](https://user-images.githubusercontent.com/8575694/164364212-6a80261b-8b77-42df-8433-f535ee4a55e6.png)

![image](https://user-images.githubusercontent.com/8575694/164364414-5ff7f12e-08ab-435a-953c-1994f85ce3a7.png)

**Comment by andrewhancox** [comment]
I'm not clear on how the translation span got injected. It's not a rich editor so it wasn't the atto plugin, looking at the db schema there is not a 'nameformat' column to match the 'name' column so the CLI tool would not have put them in...

If you let me know the question type I will take a quick look and see what is going on.

**Comment by jgramp** [comment]
Span's have somehow been inserted into the input answer fields (not Atto fields) for:

- Numerical question
- Short answer question
- missing words question

**Comment by andrewhancox** [comment]
Did you run the CLI tool to automatically add them to existing content?

**Comment by jprovasnik** [comment]
Hi - this is still causing issues on my 4.3 site - users cannot attempt exams because of span tags being inserted into non-atto fields in the following question types:  Calculated shortanswer, Calculated multichoice, Short Answer, Select missing word.  When I edit the questions and remove the spans, all is fine, until the spans get reintroduced with the scheduled task "Insert translation spans tags \filter_translations\task\insert_spans" runs again.

Any thoughts on how to define pages in which not to insert these span tags? i do have various php pages added to the plugin's "Pages to leave untranslated filter_translations | untranslatedpages" setting, but I guess that doesnt prevent the spans from being inserted on those pages too?

Student-view of Error
<img width="1010" alt="student view of attempt quiz error" src="https://github.com/andrewhancox/moodle-filter_translations/assets/44345287/d9308c2d-6195-48ac-9b0c-e8b1d7696529">

Calculated Multichoice
<img width="904" alt="calculated multichoice spans" src="https://github.com/andrewhancox/moodle-filter_translations/assets/44345287/eab80393-cfeb-4c68-8c66-f7f47bc621c8">

Calculated shortanswer
<img width="1005" alt="calculated spans" src="https://github.com/andrewhancox/moodle-filter_translations/assets/44345287/b837ba57-7089-4e25-95a4-98a4642eec69">

Select missing word:
<img width="803" alt="select missing word spans" src="https://github.com/andrewhancox/moodle-filter_translations/assets/44345287/1fad5c2e-6fc6-4599-a102-a0d9520ed7a5">

Short answer:
<img width="723" alt="short answer spans" src="https://github.com/andrewhancox/moodle-filter_translations/assets/44345287/f790743e-508f-4b66-967d-1389aa66279e">

**Comment by rjnl** [comment]
You likely have the  question "answer"  field in the filter_translations | columndefinition JSON setting.
The insert spans scheduled task will add the translation span tags to all the fields defined in this setting.
You will need to remove the answer field and any other field that should not have the span tags from the filter_translations | columndefinition setting. (Note: all the scheduled tasks for this plugin refer to the columndefinition setting.)

See this note regarding enabling [scheduled tasks](https://docs.moodle.org/403/en/Content_translation_plugin_set#Scheduled_tasks) for this plugin.

To get your quizzes to work again, you will have to manually edit and remove these translation span tags.

Hope this helps.

**Comment by jprovasnik** [comment]
Yes!! This does - I had the JSON entry below and didn't realize this was for the "Short Answer" question type.

    "question_answers": [ 
        "answer" 
        "feedback" 
    ], 

I'll go through my other entries to see which database tables/titles I've previously included in the columndefinition list that also don't have clear titles. Once I remove a few more I'll be all set.
